src/cpl/nuopc/atm_stream_ndep.F90

```diff
@@ -0,0 +1,254 @@
+    if (ierr /= PIO_NOERR) then
+       call endrun(' ERROR finding variable: '//trim(stream_varlist_ndep(1))//" in file: "// &
+            trim(stream_fldFileName_ndep)//errMsg(sourcefile, __LINE__))
+    else
+       ierr = PIO_get_att(File, vardesc, "units", ndepunits)
+    end if
+    call pio_seterrorhandling(File, err_handling)
+    call cam_pio_closefile(File)
+
+    ! Now check to make sure they are correct
+    if (.not. trim(ndepunits) == "g(N)/m2/s"  )then
+       call endrun(' ERROR in units for nitrogen deposition equal to: '//trim(ndepunits)//" not units expected"// &
+            errMsg(sourcefile, __LINE__))
+    end if
+
+  end subroutine stream_ndep_check_units
+
+  !================================================================
+  subroutine stream_ndep_interp(cam_out, rc)
+
+    use dshr_methods_mod , only : dshr_fldbun_getfldptr
+    use dshr_strdata_mod , only : shr_strdata_advance
+    use camsrfexch       , only : cam_out_t
+    use time_manager     , only : get_curr_date
+    use physics_grid     , only : columns_on_task
+
+    ! input/output variables
+    type(cam_out_t) , intent(inout)  :: cam_out(:)
+    integer         , intent(out)    :: rc
+
+    ! local variables
+    integer :: i
+    integer :: year    ! year (0, ...) for nstep+1
+    integer :: mon     ! month (1, ..., 12) for nstep+1
+    integer :: day     ! day of month (1, ..., 31) for nstep+1
+    integer :: sec     ! seconds into current date for nstep+1
+    integer :: mcdate  ! Current model date (yyyymmdd)
+    real(r8), pointer :: dataptr1d_nhx(:)
+    real(r8), pointer :: dataptr1d_noy(:)
+    !-----------------------------------------------------------------------
+
+    ! Advance sdat stream
+    call get_curr_date(year, mon, day, sec)
+    mcdate = year*10000 + mon*100 + day
+    call shr_strdata_advance(sdat_ndep, ymd=mcdate, tod=sec, logunit=iulog, istr='ndepdyn', rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+       call ESMF_Finalize(endflag=ESMF_END_ABORT)
+    end if
+
+    ! Get pointer for stream data that is time and spatially interpolated to model time and grid
+    call dshr_fldbun_getFldPtr(sdat_ndep%pstrm(1)%fldbun_model, stream_varlist_ndep(1), fldptr1=dataptr1d_nhx, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+       call ESMF_Finalize(endflag=ESMF_END_ABORT)
+    end if
+    call dshr_fldbun_getFldPtr(sdat_ndep%pstrm(1)%fldbun_model, stream_varlist_ndep(2), fldptr1=dataptr1d_noy, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+       call ESMF_Finalize(endflag=ESMF_END_ABORT)
+    end if
+
+!Un-comment once cam_out data structure has been populated -JN
+#if 0
+    do i = 1, columns_on_task
+       cam_out(c)%nhx_nitrogen_flx(i) = dataptr1d_nhx(g)
+       cam_out(c)%noy_nitrogen_flx(i) = dataptr1d_noy(g)
```

Comment:
I know these are commented out for now, but can you remove the "c" indexing?

Reply:
Sure can!  The "c" indexing has now been removed.